source/server/options_impl.cc

```diff
@@ -209,4 +216,45 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv,
     throw NoServingException();
   }
 }
+
+void OptionsImpl::parseComponentLogLevels(const std::string& component_log_levels) {
+  if (!component_log_levels.empty()) {
+    std::vector<std::string> log_levels = absl::StrSplit(component_log_levels, ',');
+    for (auto& level : log_levels) {
+      std::vector<std::string> log_name_level = absl::StrSplit(level, ':');
+      std::string log_name = log_name_level[0];
+      std::string log_level = log_name_level[1];
+      if (log_name_level.size() != 2) {
```

Comment:
this check needs to come before you pull out the values?

Reply:
Yup..Will change